sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.PartitionProcessor.cs

```diff
@@ -0,0 +1,367 @@
+			{
+				_executor = executor;
+				_singleDispatch = singleDispatch;
+				_batchCheckpointFrequency = options.BatchCheckpointFrequency;
+				_logger = logger;
+				_minimumBatchesEnabled = options.MinEventBatchSize > 1; // 1 is the default
+				StoredEventsManager = new PartitionProcessorEventsManager(maxBatchSize: options.MaxEventBatchSize, minBatchSize: options.MinEventBatchSize);
+				_maxWaitTime = options.MaxWaitTime;
+			}
+
+			public Task CloseAsync(EventProcessorHostPartition context, ProcessingStoppedReason reason)
+			{
+				// signal cancellation for any in progress executions
+				_cts.Cancel();
+				StoredEventsManager.PartitionClosing();
+
+				_logger.LogDebug(GetOperationDetails(context, $"CloseAsync, {reason.ToString()}"));
+				return Task.CompletedTask;
+			}
+
+			public Task OpenAsync(EventProcessorHostPartition context)
+			{
+				_logger.LogDebug(GetOperationDetails(context, "OpenAsync"));
+				return Task.CompletedTask;
+			}
+
+			public Task ProcessErrorAsync(EventProcessorHostPartition context, Exception error)
+			{
+				string errorDetails = $"Processing error (Partition Id: '{context.PartitionId}', Owner: '{context.Owner}', EventHubPath: '{context.EventHubPath}').";
+
+				Utility.LogException(error, errorDetails, _logger);
+
+				return Task.CompletedTask;
+			}
+
+			public async Task ProcessEventsAsync(EventProcessorHostPartition context, IEnumerable<EventData> messages, CancellationToken processingCancellationToken)
+			{
+				using (CancellationTokenSource linkedCts =
+						CancellationTokenSource.CreateLinkedTokenSource(_cts.Token, processingCancellationToken))
+				{
+					_mostRecentPartitionContext = context;
+					var events = messages == null ? Array.Empty<EventData>() : messages.ToArray();
+					EventData eventToCheckpoint = null;
+
+					int eventCount = events.Length;
+
+					if (_singleDispatch)
+					{
+						UpdateCheckpointContext(events, context);
+
+						var triggerInput = new EventHubTriggerInput
+						{
+							Events = events,
+							ProcessorPartition = context
+						};
+
+						// Single dispatch
+						for (int i = 0; i < eventCount; i++)
+						{
+							if (linkedCts.Token.IsCancellationRequested)
+							{
+								break;
+							}
+
+							EventHubTriggerInput eventHubTriggerInput = triggerInput.GetSingleEventTriggerInput(i);
+							TriggeredFunctionData input = new TriggeredFunctionData
+							{
+								TriggerValue = eventHubTriggerInput,
+								TriggerDetails = eventHubTriggerInput.GetTriggerDetails(context)
+							};
+
+							await _executor.TryExecuteAsync(input, linkedCts.Token).ConfigureAwait(false);
+							eventToCheckpoint = events[i];
+						}
+					}
+					else
+					{
+						// Batch dispatch
+
+						if (_minimumBatchesEnabled)
+						{
+							var triggerEvents = StoredEventsManager.ProcessWithStoredEvents(context, messages.ToList(), false, linkedCts.Token);
```

Comment:
The reason why I named it `ProcessWithStoredEvents` was to mirror `EventHubListener.PartitionProcessor.ProcessEvents`, since I was trying to denote, "do the same thing but add in stored events"